### ISSUE TITLE
Some level of test automation

### ISSUE THREAD

**Issue by fdion** [adamerose/PandasGUI#144]
I think it would be beneficial to set up some github actions to do some basic checks.

In particular, it would be good to test across python 3.7-3.9, Ubuntu 18LTS, 20LTS using miniconda and make sure the app can at least be launched (using xvfb).

I'm thinking something akin to what I do for my open source projects like Hotelling:

https://github.com/dionresearch/hotelling/blob/master/.github/workflows/dev.yml

but with an added extra step of integration test (although a QT app, it can be run headless with xvfb), to make sure no packages are missing from the requirements, no compatibility issues from python version etc.

**Comment by adamerose** [comment]
Yes I agree. Once we have CI set up that's probably when I'll call the next release 1.0.0 

I haven't done any work on this yet just planning, I'll take a look at your example
